openmdao/core/driver.py

```diff
@@ -93,6 +95,38 @@ def _setup(self, root):
         self._objs = objs
         self._cons = cons
 
+        # Cache scalers for derivative calculation
+
+        for name, meta in iteritems(desvars):
+            scaler = meta.get('scaler')
+            if isinstance(scaler, np.ndarray):
+                if all(scaler == 1.0):
+                    continue
+            elif scaler == 1.0:
+                continue
+
+            self.dv_conversions[name] = np.reciprocal(scaler)
+
+        for name, meta in iteritems(objs):
+            scaler = meta.get('scaler')
+            if isinstance(scaler, np.ndarray):
+                if all(scaler == 1.0):
+                    continue
```

Comment:
the "array all == 1.0" case **seems to be** a hole in coverage as well.. for both obj and con (line 124)